opentelemetry-api/src/opentelemetry/util/__init__.py

```diff
@@ -33,18 +34,39 @@ def time_ns() -> int:
         return int(time.time() * 1e9)
 
 
-def _load_provider(provider: str) -> Union["TracerProvider", "MeterProvider"]:  # type: ignore
+def _load_provider(
+    provider: str,
+) -> Union["TracerProvider", "MeterProvider"]:  # type: ignore
     try:
         return next(  # type: ignore
             iter_entry_points(
                 "opentelemetry_{}".format(provider),
-                name=getattr(  # type: ignore
-                    Configuration(), provider, "default_{}".format(provider),  # type: ignore
+                name=getattr(
+                    Configuration(),  # type: ignore
+                    provider,
+                    "default_{}".format(provider),
                 ),
             )
         ).load()()
     except Exception:  # pylint: disable=broad-except
-        logger.error(
-            "Failed to load configured provider %s", provider,
-        )
+        logger.error("Failed to load configured provider %s", provider)
         raise
+
+
+# Pattern for matching up until the first '/' after the 'https://' part.
+URL_PATTERN = "^(https?|ftp):\\/\\/.*\\/"
```

Comment:
This does not match up until the first `/` after the `https://` part:

```python
In [9]: URL_PATTERN
Out[9]: '^(https?|fpt):\\/\\/.*\\/'

In [10]: match(URL_PATTERN, "https://something/something/something/").group()
Out[10]: 'https://something/something/something/'
```
The `*` operator is greedy and will match everything it can including `/`.
```suggestion
_URL_PATTERN = r"(https?|ftp)://.*?/"
```
```python
In [19]: URL_PATTERN = r"(https?|ftp)://.*?/"

In [20]: match(URL_PATTERN, "https://something/something/something/").group()
Out[20]: 'https://something/'
```
Also, [match](https://docs.python.org/3/library/re.html#re.match) starts matching from the beginning of the string, there is no need to use `^`.
Even better, don't declare a module-level constant if its value is to be used only once in the code.

Reply:
Fixed.